testcases/open_posix_testsuite/conformance/interfaces/strcpy/1-1.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is licensed under the GPL license. For the full content
+ * of this license, see the COPYING file at the top level of this
+ * source tree.
+ */
+
+/*
+ * assertion:
+ *	The function will copy the string pointed to by s1 (including the
+ *	terminating NUL character) into the array pointed to by s2. And
+ *	it will return s2.
+ *
+ * method:
+ *	-Generate sample string s1 of variable lengths in a loop.
+ *	-Use strcpy() to copy s1 into sample string s2 in each iteration.
+ *	-Compare both strings(s1 and s2).
+ *	-Also compare returned pointer with s2.
+*/
+
+#define _XOPEN_SOURCE 700
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <time.h>
+#include "posixtest.h"
+
+#define TNAME "strcpy/1-1.c"
+
+char *random_string(int len) {
+	char *input_string = "01234567890\n\t\r\v\fabcdefghijklmnopqrstuvwxyz~!#$%^&*()_+ABCDEF"
+							"GHIJKLMNOPQRSTUVWXYZ";
+	size_t string_len = strlen(input_string);
```

Comment:
Why do you call strlen() on a statically allocated string? For this kind of strings strlen(str) == sizeof(str) - 1.

Reply:
Same reason, instead of hardcoding the length of string or defining a new macro for this, i am using strlen() to get it.